File: leetcode/problems/language-specific/go/268.missing-number.go
/*
 * @lc app=leetcode id=268 lang=golang
 *
 * [268] Missing Number
 */

// @lc code=start

// * Math Solution | O(n) Time | O(1) Space

func missingNumber(nums []int) int {
	numsSize := len(nums)
	total, sum := (numsSize*(numsSize+1))/2, 0
	for _, v := range nums {
		sum += v
	}

	return total - sum
}

// @lc code=end

File: leetcode/problems/language-specific/python/268.missing-number.py
#
# @lc app=leetcode id=268 lang=python3
#
# [268] Missing Number
#

# @lc code=start

# * Math Solution | O(n) Time | O(1) Space
# * n -> The length of nums array


class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        n = len(nums)
        return ((n * (n + 1)) // 2) - sum(nums)


# @lc code=end
